controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -328,3 +343,42 @@ func addStorageRelatedAnnotations(obj client.Object, storageConsumerName, storag
 	annotations[StorageRequestAnnotation] = storageRequest
 	annotations[StorageCephUserTypeAnnotation] = cephUserType
 }
+
+func (r *StorageConsumerReconciler) reconcileMaintenanceMode() error {
+
+	rbdMirrorDeployments := &appsv1.DeploymentList{}
+	err := r.List(r.ctx, rbdMirrorDeployments, client.InNamespace(r.storageConsumer.Namespace), client.MatchingLabels{"app": "rook-ceph-rbd-mirror"}, client.Limit(1))
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if errors.IsNotFound(err) {
+		return nil
+	}
+
+	replicas := int32(1)
+	if enableMaintenanceMode(r.storageConsumerList) {
```

Comment:
lets fetch rbdMirrorDeployments only if we need to scale down the deployment.

Reply:
I am following this:
1. Fetch rbdMirror, if no rbdMirror found return 
2. Set replicas to 1
3. If enableMaintenanceMode, set replicas to 0
4. Update the Deployment with the desired replica